js/src/edit-block/components/editor-provider.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useLayoutEffect } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+// @ts-ignore type declaration not available
+import { EntityProvider } from '@wordpress/core-data';
+// @ts-ignore type declaration not available
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { useEditor } from '../hooks';
+
+/**
+ * @typedef {Object} EditorProviderProps The props of the component.
+ * @property {Object} post The post for the editor.
+ * @property {Object} settings The editor settings.
+ * @property {React.ReactElement} children The component's children.
+ */
+
+/**
+ * The editor provider, forked from Gutenberg.
+ *
+ * This is forked so we can remove the <BlockEditorProvider>.
+ * That adds .blocks to the editedEntityRecord,
+ * which causes getEditedPostContent() to return ''.
+ *
```

Comment:
@kienstra Do you see any risks in doing this that we should watch out for?

Reply:
Yeah, if [getEditedPostContent()](https://github.com/WordPress/gutenberg/blob/ca6a2263663a27504b95bd8e2a97c8796250c8f9/packages/editor/src/store/selectors.js#L982) is ever changed so that it doesn't return `record.content`, that'd be a problem:

```
} else if ( record.content ) {
	return record.content;
}
```

In that case, we'd probably have to implement these components on our own, and our own simplified version of the `'core/editor'` store:

![core-editor-store](https://user-images.githubusercontent.com/4063887/102381914-6dcda900-3f8f-11eb-818a-1e7c1f44c022.png)

It'd be possible, just a lot of code.

So I think we'll have to keep testing with Gutenberg to make sure the next version of WP won't break.

But for the moment, I think we get a lot from using Gutenberg's `'core/editor'` store.

Reply:
Sounds good. Thanks!